Source/genetic_factory.cpp

```diff
@@ -1,21 +1,79 @@
 #include "genetic_factory.h"
 
-chromosome* genetic_factory::create_chromosome(chromosome_metadata* chromosome_metadata) {
-	return nullptr;
+genetic_factory& genetic_factory::get_instance() {
+    static genetic_factory instance;
+    return instance;
+}
+
+//region plugin registration
+
+void genetic_factory::register_selection_algorithm(const try_create_selection_algorithm_t try_create) {
+    _registered_selection_algorithms.push_back(try_create);
```

Comment:
What does `push_back` do?

Reply:
Adds an element to the back of a std::vectory